vissl/models/base_ssl_model.py

```diff
@@ -551,11 +551,11 @@ def init_model_from_weights_params_file(
     def is_clustering_model(self):
         """
         Whether or not the model is a clustering model
-        (only supports SwAV for the moment)
+        (supports SwAV and DeepCluster_v2 for the moment)
         """
-        return len(self.heads) == 1 and any(
-            isinstance(module, SwAVPrototypesHead) for module in self.heads[0].modules()
-        )
+        return len(self.heads) == 1 or any(
+            isinstance(module, MLP) for module in self.heads[0].modules()) or any(
```

Comment:
So I don't think any model with an MLP head should return true for `is_clustering_model`. 

Since it's difficult to find a way to tell whether or not the model is DeepCluster in `base_ssl_model.py` -- I would recommend moving #is_clustering_model to SelfSupervisionTask `train_task.py`. 

So `trainer_main.py` reads: 

```
        error_message = "Extracting clusters is only available for pre-training methods based on clusters"
        assert self.task.is_clustering_model(), error_message
```

You can then use `task.loss.loss_config['name'] == 'deepclusterv2_loss'` as a requirement. LMK if that makes sense.